config/configLoadHelpers.go

```diff
@@ -391,42 +391,42 @@ func loadConfigsIntoMap(dest map[string]any, configs []configData) error {
 // validateConfigs gets the configs from the given location and validates them.
 // It returns a list of failures; if the list is empty, the config is valid.
 // err is non-nil only for significant errors like a missing file.
-func validateConfigs(configs []configData, opts *CmdEnv) ([]string, error) {
+func validateConfigs(configs []configData, opts *CmdEnv, currentVersion ...string) (warnings []string, errors []string, err error) {
```

Comment:
I'm really not fond of this change of signature. Instead of having two buckets of warnings and errors, I think it would be more useful to change the return type from []string. Maybe something like this?

```go
type ValidationResult {
  Message string
  Severity int    // an enum with Warning and Error at least
}
type ValidationResults []ValidationResult
func (v ValidationResults) isOnlyWarnings() {...}
```